### ISSUE TITLE
Manually unmasking the brighter image leads to strong color artifacts

### ISSUE THREAD

**Issue by Beep6581** [jcelaya/hdrmerge#64]
When I manually unmask an area where I want a darker image to be used, the result is correct. But when I manually unmask an area for the lighter image to show through, that area becomes cyan or some other very wrong color, instead of being clipped white. In other words, the white level is wrong, because it turns out grayish instead of being clipped white.

Original mask automatically made: http://i.imgur.com/ueO1DgU.png
The image looks fine in RawTherapee: http://i.imgur.com/LzPdpqX.png
Manually painted mask to use the darker image: http://i.imgur.com/XC2LSzn.png
Still looks good: http://i.imgur.com/XC2LSzn.png
But if I pain the mask to use the lighter image: http://i.imgur.com/DU3A4zO.png
The result is bad: http://i.imgur.com/uXrgJYI.png

This has been the case probably since I started using HDRMerge a year ago, I can't imagine why I haven't reported it earlier - I certainly meant to.

**Comment by jcelaya** [comment]
Yes... Well, this should not be a bug, it's just how it works. To simplify it, we can say that for every pixel in the set of images, HDRMerge takes the most exposed one that is not saturated. If you unmask a darker image, those pixels are less exposed. They basically contain the same information, but with more noise. However, if you unmask a lighter image, those pixels will be saturated. They do not contain any useful information, they are blown out. Useless, don't do that.

However, at least, those blown out areas you just revealed should be gray (not white, because the brightness of the lighter images is adjusted to match the darker ones). Cyan and pink color artifacts appear because there is no highlight recovery algorithm (AFAIK, these algorithms are applied only after interpolating the raw pixels). And, if the mask radius is a bit large, these saturated pixels may get not revealed, but mixed with other useful pixels, producing these color artifacts even if they seem to be masked.

I have fixed this last problem in the develop branch. However, as I said before, you should really never unmask a lighter image. If you find yourself doing that often, maybe you should merge more shots in every HDR DNG (5 or 7 instead of 3, for instance, or take two consecutive sets of 3 shots)
